src/main/java/edu/cornell/kfs/module/purap/rest/resource/CuEinvoiceApiResource.java

```diff
@@ -0,0 +1,248 @@
+package edu.cornell.kfs.module.purap.rest.resource;
+
+import com.google.gson.Gson;
+import edu.cornell.kfs.module.purap.CUPurapConstants;
+import edu.cornell.kfs.module.purap.dataaccess.CuEinvoiceDao;
+import edu.cornell.kfs.module.purap.dataaccess.impl.CuEinvoiceDaoOjb;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.kuali.kfs.krad.dao.LookupDao;
+import org.kuali.kfs.krad.util.KRADPropertyConstants;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.purap.PurapPropertyConstants;
+import org.kuali.kfs.module.purap.businessobject.PurApItem;
+import org.kuali.kfs.module.purap.document.PurchaseOrderDocument;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.vnd.VendorConstants;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+@Path("api")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class CuEinvoiceApiResource {
+
+    private static final Log LOG = LogFactory.getLog(CuEinvoiceApiResource.class);
+
+    private LookupDao lookupDao;
+    private CuEinvoiceDao cuEinvoiceDao;
+    private Gson gson = new Gson();
+
+    @Context
+    protected HttpServletRequest servletRequest;
+
+    @Context
+    protected HttpServletResponse servletResponse;
+
+    @GET
+    public Response describeEinvoiceApiResource() {
+        return Response.ok(CUPurapConstants.Einvoice.EINVOICE_KFS_API_DESCRIPTION).build();
+    }
+
+    @GET
+    @Path("vendors")
+    public Response getVendor(@Context HttpHeaders headers) {
+        try {
+            List<String> vendorNumbers = getVendorNumbersFromRequest();
+            List<VendorDetail> vendors = getCuEinvoiceDao().getVendors(vendorNumbers);
+            List<Properties> vendorsSerialized = vendors.stream().map(vendor -> getVendorProperties(vendor)).collect(Collectors.toList());
+            return Response.ok(gson.toJson(vendorsSerialized)).build();
+        } catch (BadRequestException ex) {
+            LOG.error(ex);
+            return respondGetVendorsBadRequest();
```

Comment:
Thanks for the clarification,  It looks like many of these functions are doing the same error handling.  To consolidate, we could just catch exception in these functions, and in the catch call an error handling function that could include an instanceof check so it could do different things if the exception is BadRequestException or whatever else.  Also, the log statements should include a function name

Reply:
I have consolidated into 1 catch block and using instance of in a ternary statement.